### ISSUE TITLE
[Bug]: Unable to obtain resource path from Resource method symbol in an object

### ISSUE THREAD

**Issue by MohamedSabthar** [ballerina-platform/ballerina-lang#37568]
### Description

```ballerina
public type Person distinct service object  {
    isolated resource function get name() returns string;
};
```

Unable to obtain resource path from the above resource function

### Steps to Reproduce

Taking `Person` as the `symbol` and accessing the resource path in the following way 
```java
((ResourceMethodSymbol)((ObjectTypeSymbol) ((TypeDefinitionSymbol) symbol).typeDescriptor()).methods().values().stream().findFirst().get()).resourcePath()
```

 resulting in a `ClassCastException`

### Labels

CompilerFE

**Comment by rdulmina** [comment]
Can u put the stack trace

**Comment by MohamedSabthar** [comment]
Stack trace: 

java.lang.ClassCastException: class org.wso2.ballerinalang.compiler.semantics.model.symbols.BObjectTypeSymbol cannot be cast to class org.wso2.ballerinalang.compiler.semantics.model.symbols.BClassSymbol (org.wso2.ballerinalang.compiler.semantics.model.symbols.BObjectTypeSymbol and org.wso2.ballerinalang.compiler.semantics.model.symbols.BClassSymbol are in unnamed module of loader 'app')
	at io.ballerina.compiler.api.impl.symbols.BallerinaResourceMethodSymbol.resourcePath(BallerinaResourceMethodSymbol.java:72)
	at io.ballerina.stdlib.graphql.compiler.service.InterfaceFinder.populateInterfaces(InterfaceFinder.java:63)
	at io.ballerina.stdlib.graphql.compiler.ServiceAnalysisTask.perform(ServiceAnalysisTask.java:60)
	at io.ballerina.stdlib.graphql.compiler.ServiceAnalysisTask.perform(ServiceAnalysisTask.java:40)
	at io.ballerina.projects.SyntaxNodeAnalysisTask.perform(SyntaxNodeAnalysisTask.java:46)
	at io.ballerina.projects.SyntaxTreeVisitor.runAnalysisTasks(SyntaxTreeVisitor.java:93)
	at io.ballerina.projects.SyntaxTreeVisitor.visitSyntaxNode(SyntaxTreeVisitor.java:80)
	at io.ballerina.compiler.syntax.tree.NodeVisitor.visit(NodeVisitor.java:63)
	at io.ballerina.compiler.syntax.tree.ServiceDeclarationNode.accept(ServiceDeclarationNode.java:78)
	at io.ballerina.projects.SyntaxTreeVisitor.visitSyntaxNode(SyntaxTreeVisitor.java:85)
	at io.ballerina.compiler.syntax.tree.NodeVisitor.visit(NodeVisitor.java:43)
	at io.ballerina.projects.SyntaxTreeVisitor.runAnalysisTasks(SyntaxTreeVisitor.java:68)
	at io.ballerina.projects.SyntaxNodeAnalysisTaskRunner.runTasks(SyntaxNodeAnalysisTaskRunner.java:76)
	at io.ballerina.projects.SyntaxNodeAnalysisTaskRunner.runTasks(SyntaxNodeAnalysisTaskRunner.java:59)
	at io.ballerina.projects.SyntaxNodeAnalysisTaskRunner.runTasks(SyntaxNodeAnalysisTaskRunner.java:51)
	at io.ballerina.projects.CodeModifierManager.runSyntaxNodeAnalysisTasks(CodeModifierManager.java:115)
	at io.ballerina.projects.CodeModifierManager.getCodeModifyTaskResult(CodeModifierManager.java:79)
	at io.ballerina.projects.CodeModifierManager.runCodeModifiers(CodeModifierManager.java:57)
	at io.ballerina.projects.Package.runCodeGenAndModifyPlugins(Package.java:269)
	at io.ballerina.stdlib.graphql.compiler.ServiceValidationTest.getDiagnosticResult(ServiceValidationTest.java:910)
	at io.ballerina.stdlib.graphql.compiler.ServiceValidationTest.testInterfaces(ServiceValidationTest.java:169)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
	at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
	at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
	at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
	at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
	at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.testng.TestRunner.privateRun(TestRunner.java:794)
	at org.testng.TestRunner.run(TestRunner.java:596)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
	at org.testng.SuiteRunner.run(SuiteRunner.java:276)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
	at org.testng.TestNG.runSuites(TestNG.java:1063)
	at org.testng.TestNG.run(TestNG.java:1031)
	at org.gradle.api.internal.tasks.testing.testng.TestNGTestClassProcessor.runTests(TestNGTestClassProcessor.java:141)
	at org.gradle.api.internal.tasks.testing.testng.TestNGTestClassProcessor.stop(TestNGTestClassProcessor.java:90)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:61)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.stop(TestWorker.java:135)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.base/java.lang.Thread.run(Thread.java:829)

**Comment by dulajdilshan** [comment]
This was fixed via https://github.com/ballerina-platform/ballerina-lang/pull/37575. @MohamedSabthar Can you please verify?

**Comment by MohamedSabthar** [comment]
> This was fixed via #37575. @MohamedSabthar Can you please verify?

@dulajdilshan, fix works
